websites/A/Animesuge/presence.ts

```diff
@@ -0,0 +1,152 @@
+    "/ova": "Searching for OVA animes",
+    "/ona": "Searching for ONA animes",
+    "/special": "Searching for special anime episodes",
+    "/search": "Searching anime by name",
+    "/az-list": "Seaching all animes",
+    "/most-watched": "Searching most watched animes",
+    "/upcoming": "Searching upcoming animes",
+    "/faq": "Reading the FAQ",
+    "/contact": "Reading the contacts",
+    "/user/settings": "Changing the settings",
+    "/user/watchlist": "Looking at their watchlist",
+    "/user/import": "Importing their MAL list to Animesuge!"
+  };
+
+let endTime: number,
+  currentTime: number;
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+
+presence.on("iFrameData", async (data: {endTime: number; currentTime: number;}) => {
+  ({endTime, currentTime} = data);
+});
+
+presence.on("UpdateData", async () => {
+  const page = document.location.pathname,
+    animeName = document.querySelector(
+      "#body > div > div > div > div > section > div > h1"
+    ) as HTMLElement,
+    data: PresenceData = {
+      largeImageKey: "animesuge"
+    },
+    search: URLSearchParams = new URLSearchParams(document.location.search.substring(1));
+  if (page === "/") {
+    data.details = "At the home page";
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page.includes("/anime")) {
+    data.details = pages["/anime"];
+    data.state = `Watching ${animeName.textContent}`;
+    [data.startTimestamp, data.endTimestamp] = getTimestamps(currentTime, endTime);
+    data.buttons = [
+      {
+        label: "Watch it too!",
+        url: `http://animesuge.io${page}`
+      }
+    ];
+  } else if (page.includes("/genre")) {
+    const genre = page.slice("/genre/".length);
+    data.details = pages["/genre"];
+    data.state = `Searching for ${genre.charAt(0).toUpperCase() + genre.slice(1)} Animes`;
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/newest") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/updated") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/ongoing") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/added") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/tv") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
```

Comment:
you can just make 1 timestamp and set it at the start, this is overkill doing it 9000+ times